google/cloud/storage/bucket.py

```diff
@@ -2236,6 +2237,130 @@ def rename_blob(
             )
         return new_blob
 
+    @create_trace_span(name="Storage.Bucket.moveBlob")
+    def move_blob(
+        self,
+        blob,
+        new_name,
+        client=None,
+        if_generation_match=None,
+        if_generation_not_match=None,
+        if_metageneration_match=None,
+        if_metageneration_not_match=None,
+        if_source_generation_match=None,
+        if_source_generation_not_match=None,
+        if_source_metageneration_match=None,
+        if_source_metageneration_not_match=None,
+        timeout=_DEFAULT_TIMEOUT,
+        retry=DEFAULT_RETRY,
```

Comment:
Could you check the retry policy? Looking at copy operation and other language implementations, move object appears to be conditionally idempotent with ifGenerationMatch.

Reply:
You're right, for some reason I got the wrong idea about other language impls. Fixed. Nice catch!